ksql-engine/src/test/java/io/confluent/ksql/codegen/CodeGenRunnerTest.java

```diff
@@ -63,9 +112,288 @@ private Analysis analyzeQuery(String queryStr) {
+        String simpleQuery = String.format(queryFormat, cola, colb);
+        Analysis analysis = analyzeQuery(simpleQuery);
+
+        ExpressionMetadata expressionEvaluatorMetadata0 = codeGenRunner.buildCodeGenFromParseTree
+            (analysis.getSelectExpressions().get(0));
+        Assert.assertTrue(expressionEvaluatorMetadata0.getIndexes().length == 2);
+        int idx0 = expressionEvaluatorMetadata0.getIndexes()[0];
+        int idx1 = expressionEvaluatorMetadata0.getIndexes()[1];
+        Assert.assertThat(idx0, anyOf(equalTo(cola), equalTo(colb)));
+        Assert.assertThat(idx1, anyOf(equalTo(cola), equalTo(colb)));
+        Assert.assertNotEquals(idx0, idx1);
+        if (idx0 == colb) {
+            Object tmp = values[0];
+            values[0] = values[1];
+            values[1] = tmp;
+        }
+        Assert.assertEquals(expressionEvaluatorMetadata0.getUdfs().length, 2);
+        Object result0 = expressionEvaluatorMetadata0.getExpressionEvaluator().evaluate(values);
+        assertThat(result0, instanceOf(Boolean.class));
+        return (Boolean)result0;
+    }
+
+    private boolean evalBooleanExprEq(int cola, int colb, Object values[]) throws Exception {
+        return evalBooleanExpr("SELECT col%d = col%d FROM CODEGEN_TEST;", cola, colb, values);
+    }
+
+    private boolean evalBooleanExprNeq(int cola, int colb, Object values[]) throws Exception {
+        return evalBooleanExpr("SELECT col%d != col%d FROM CODEGEN_TEST;", cola, colb, values);
+    }
+
+    private boolean evalBooleanExprIsDistinctFrom(int cola, int colb, Object values[]) throws Exception {
+        return evalBooleanExpr("SELECT col%d IS DISTINCT FROM col%d FROM CODEGEN_TEST;", cola, colb, values);
+    }
+
+    private boolean evalBooleanExprLessThan(int cola, int colb, Object values[]) throws Exception {
+        return evalBooleanExpr("SELECT col%d < col%d FROM CODEGEN_TEST;", cola, colb, values);
+    }
+
+
+    private boolean evalBooleanExprLessThanEq(int cola, int colb, Object values[]) throws Exception {
+        return evalBooleanExpr("SELECT col%d <= col%d FROM CODEGEN_TEST;", cola, colb, values);
+    }
+
+    private boolean evalBooleanExprGreaterThan(int cola, int colb, Object values[]) throws Exception {
+        return evalBooleanExpr("SELECT col%d > col%d FROM CODEGEN_TEST;", cola, colb, values);
+    }
+
+    private boolean evalBooleanExprGreaterThanEq(int cola, int colb, Object values[]) throws Exception {
+        return evalBooleanExpr("SELECT col%d >= col%d FROM CODEGEN_TEST;", cola, colb, values);
+    }
+
+    @Test
+    public void testNullEquals() throws Exception {
+        Assert.assertFalse(evalBooleanExprEq(5, 0, new Object[]{null, 12344L}));
+        Assert.assertFalse(evalBooleanExprEq(5, 0, new Object[]{null, null}));
+    }
+
+    @Test
+    public void testIsDistinctFrom() throws Exception {
+        Assert.assertFalse(evalBooleanExprIsDistinctFrom(5, 0, new Object[]{12344, 12344L}));
```

Comment:
Might make the code more readable if the `(5, 0, ...` magic numbers were pulled out into constants, e.g. `(INT32_FIELD, INT64_FIELD, ...`